core/tests/protractor/accessibility.js

```diff
@@ -41,6 +120,164 @@ describe('screenreader and keyboard user accessibility features', function() {
         .getAttribute('id'));
   });
 
+  it('should move focus to the skip button in the library page',
+    async function() {
+      // Should move the focus to the skip to main content button.
+      await libraryPage.get();
+      await browser.actions().sendKeys('s').perform();
+      var skipButton = element(by.css('.protractor-test-skip-link'));
+      expect(await skipButton.getAttribute('id')).toEqual(
+        await (await browser.driver.switchTo().activeElement())
+          .getAttribute('id'));
+
+      // Should move the focus away from the skip to main content button.
+      await browser.actions().sendKeys(protractor.Key.TAB).perform();
+      await browser.actions().sendKeys(protractor.Key.TAB).perform();
+      expect(await skipButton.getAttribute('id')).not.toEqual(
+        await (await browser.driver.switchTo().activeElement())
+          .getAttribute('id'));
+
+      // Should move the focus to the skip to main content button.
+      await browser.actions().sendKeys('s').perform();
+      expect(await skipButton.getAttribute('id')).toEqual(
+        await (await browser.driver.switchTo().activeElement())
+          .getAttribute('id'));
+    });
+
+  it('should move focus to the search bar in the library page',
+    async function() {
+      // Should move the focus to the search bar.
+      await libraryPage.get();
+      await browser.actions().sendKeys('/').perform();
+      var searchBar = element(by.css('.protractor-test-search-input'));
+      expect(await searchBar.getAttribute('id')).toEqual(
+        await (await browser.driver.switchTo().activeElement())
+          .getAttribute('id'));
+
+      // Should move the focus away from the search bar.
+      await browser.actions().sendKeys(protractor.Key.TAB).perform();
+      await browser.actions().sendKeys(protractor.Key.TAB).perform();
+      expect(await searchBar.getAttribute('id')).not.toEqual(
+        await (await browser.driver.switchTo().activeElement())
+          .getAttribute('id'));
+
+      // Should move the focus to the search bar.
+      await browser.actions().sendKeys('/').perform();
+      expect(await searchBar.getAttribute('id')).toEqual(
+        await (await browser.driver.switchTo().activeElement())
+          .getAttribute('id'));
+    });
+
+  it('should move focus to the category bar in library page',
+    async function() {
+      // Should move the focus to the category bar.
+      await libraryPage.get();
```

Comment:
Can you add waitFor statements to check that the elements on this page have loaded? Elements can still be loading when the loading indicator disappears, so it's probably better to check for the specific elements you need. Ditto throughout this file.